spec/features/user_show_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+require 'spec_helper'
+
+RSpec.describe 'Users Show', type: :request do
+  subject { page }
+
+  before(:each) do
+    User.create(id: 1, name: 'Tom', post_counter: 5,
+                photo: '',
+                bio: 'A Full Stack Developer')
+    Post.create(title: 'Post 1 by Tom', text: 'This post is written by tom', comment_counter: 0, like_counter: 0,
+                author_id: 1)
+    Post.create(title: 'Post 2 by Tom', text: 'This post is written by tom', comment_counter: 0, like_counter: 0,
+                author_id: 1)
+    Post.create(title: 'Post 3 by Tom', text: 'This post is written by tom', comment_counter: 0, like_counter: 0,
+                author_id: 1)
+    Post.create(title: 'Post 4 by Tom', text: 'This post is written by tom', comment_counter: 0, like_counter: 0,
+                author_id: 1)
+    Post.create(title: 'Post 5 by Tom', text: 'This post is written by tom', comment_counter: 0, like_counter: 0,
+                author_id: 1)
+  end
+  it 'renders the name of the user' do
+    get user_path(1)
+    expect(response.body).to include('Tom')
+  end
+
+  it 'renders the bio of the user' do
+    get user_path(1)
+    expect(response.body).to include('A Full Stack Developer')
+  end
+  it 'renders the number of posts of the user' do
+    get user_path(1)
+    expect(response.body).to include('5')
+  end
+  it 'renders the profile picture of the user' do
+    get user_path(1)
+    expect(response.body).to include('img')
```

Comment:
- [ ] kindly add Tests that redirect to the post's show page when you click on a user's post and test that redirects to the user's post's index page when you click on a **see all posts**

Reply:
Fixed.